examples/n-ary-app.kl

```diff
@@ -59,6 +59,30 @@
       [[list [_ args body]]
        [pure [list-syntax ['my-lam args body] stx]]])]])]
 
+(define-macros
+  ([flet*
+    (lambda (stx)
+      (syntax-case stx
+        [(list (_ fun body))
+         (syntax-case fun
+           [(list (name args def))
+            (syntax-case args
+              [(cons arg more-args)
+               (pure (list-syntax
+                      ('flet
+                       (list-syntax (name
+                                     (list-syntax (arg) stx)
+                                     (list-syntax ('lambda more-args def) stx))
+                                    stx)
```

Comment:
We're pretty much always using `stx` for the last argument of `list-syntax` and `cons-list-syntax`; it seems like this argument complicates the API without bringing much value. Now that we have `replace-loc`, maybe we can simplify that API in the common case where we don't want to think much about this third argument, and use `replace-loc` in the few places where the location matters?

Reply:
I think that's a reasonable idea, but we might want to tackle it in a separate PR

Reply:
The last `stx` argument is mostly for the scopes, and only incidentally for the source location. It's a bit repetitive, but the point is to say "the identifiers in the resulting syntax object should be understood as originating in *this* context, unless otherwise specified on the identifier itself", not to say "it comes from this file and line".